scalding-parquet-scrooge/src/main/scala/com/twitter/scalding/parquet/scrooge/ParquetCollectionFormatForwardCompatibility.scala

```diff
@@ -0,0 +1,153 @@
+   * be maintained in the formatted result.
+   */
+  private def formatForwardCompatibleType(sourceType: Type, targetType: Type): Type = {
+    (unwrapGroup(sourceType), unwrapGroup(targetType)) match {
+      case _ if sourceType.isPrimitive || targetType.isPrimitive =>
+        // Base case
+        sourceType
+      case (
+        GroupUnwrapped(sourceWrappers, Some(sourceRepeatedListType), None),
+        GroupUnwrapped(_, Some(targetRepeatedListType), None)
+      ) =>
+        // Format list
+        val sourceRule = ParquetListFormatRule.findFirstListRule(sourceRepeatedListType, Source)
+        val targetRule = ParquetListFormatRule.findFirstListRule(targetRepeatedListType, Target)
+
+        val formattedRepeated = (sourceRule, targetRule) match {
+          case (Some(sourceRule), Some(targetRule)) => {
+            val sourceElement = sourceRule.elementType(sourceRepeatedListType)
+            val targetElement = targetRule.elementType(targetRepeatedListType)
+            // Recurse on the element instead of `repeated` type because list still can have
+            // different formats at repeated type
+            val forwardCompatElement = formatForwardCompatibleType(sourceElement, targetElement)
+            // Wrap the solved element with current source structure, and do actual conversion work
+            val forwardCompatRepeated = ParquetListFormatRule.wrapElementAsRepeatedType(
+              sourceRule,
+              sourceRepeatedListType,
+              forwardCompatElement
+            )
+            ParquetListFormatRule.formatForwardCompatibleRepeatedType(
+              forwardCompatRepeated,
+              targetRepeatedListType)
+          }
+          case _ => sourceRepeatedListType // No-op
+        }
+        // Wrapped the formatted repeated type in its original groups,
+        // describing field name and whether it's optional/required
+        sourceWrappers.foldRight(formattedRepeated) {
+          (wrapper, group) => wrapper.withNewFields(group)
+        }
+      case (
+        GroupUnwrapped(sourceWrappers, None, Some(sourceRepeatedMapType)),
+        GroupUnwrapped(_, None, Some(targetRepeatedMapType))
+      ) =>
+        // Format map
+        val forwardCompatRepeated = formatForwardCompatibleType(sourceRepeatedMapType, targetRepeatedMapType)
+        val formattedRepeated = ParquetMapFormatRule.formatForwardCompatibleRepeatedType(
+          forwardCompatRepeated,
+          targetRepeatedMapType
+        )
+        // Wrapped the formatted repeated type in its original groups,
+        // describing field name and whether it's optional/required
+        sourceWrappers.foldRight(formattedRepeated) {
+          (wrapper, group) => wrapper.withNewFields(group)
+        }
+      case _ =>
+        // Field projection
+        val sourceGroup = sourceType.asGroupType
+        val targetGroup = targetType.asGroupType
+
+        import scala.collection.JavaConverters._
```

Comment:
Import at the top of the file? That should be safe because this is `Converters` (not `Conversions`, which is much more implicit).